module/VuFind/src/VuFind/Recommend/EDSDatabases.php

```diff
@@ -0,0 +1,174 @@
+ * @package  Recommendations
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:recommendation_modules Wiki
+ */
+class EDSDatabases implements RecommendInterface
+{
+    use \VuFind\Cache\CacheTrait;
+
+    /**
+     * Results object
+     *
+     * @var \VuFind\Search\Base\Results
+     */
+    protected $results;
+
+    /**
+     * LibGuides connector
+     *
+     * @var LibGuides
+     */
+    protected $libGuides;
+
+    /**
+     * Constructor
+     *
+     * @param LibGuides    $libGuides LibGuides API connection
+     * @param Config       $config    LibGuides API configuration object
+     * @param CacheAdapter $cache     Object cache
+     */
+    public function __construct(
+        LibGuides $libGuides,
+        Config $config,
+        CacheAdapter $cache
+    ) {
+        $this->libGuides = $libGuides;
+        $this->setCacheStorage($cache);
+
+        // Cache the data related to profiles for up to 10 minutes:
+        $this->cacheLifetime = intval($config->GetAZ->cache_lifetime ?? 600);
+    }
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called before the Search Results object performs its main search
+     * (specifically, in response to \VuFind\Search\SearchRunner::EVENT_CONFIGURED).
+     * This method is responsible for setting search parameters needed by the
+     * recommendation module and for reading any existing search parameters that may
+     * be needed.
+     *
+     * @param \VuFind\Search\Base\Params $params  Search parameter object
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function init($params, $request)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called after the Search Results object has performed its main search. This
+     * may be used to extract necessary information from the Search Results object
+     * or to perform completely unrelated processing.
+     *
+     * @param \VuFind\Search\Base\Results $results Search results object
+     *
+     * @return void
+     */
+    public function process($results)
+    {
+        $this->results = $results;
+    }
+
+    /**
+     * Get terms related to the query.
+     *
+     * @return array
+     */
+    public function getResults()
+    {
+        $records = $this->results->getResults();
+        $nameToDatabase = $this->getLibGuidesDatabases();
+        $databases = [];
+        foreach ($records as $record) {
+            $databaseInfo = $nameToDatabase[$record->getDbLabel()] ?? null;
```

Comment:
Doesn't EDS support the database as a facet? If so, would it be better to use the facet data instead of iterating through records? Then you could recommend the most commonly-matched databases in the result set, instead of the potentially more arbitrary "whatever shows up on the current page."

Reply:
It does support database as a facet ("source" I think), and yes that would probably give better results.  But I believe that would require actually enabling that facet (and facets in the first place) for this recommendation module to get data from it -- which you almost definitely would not want to do, since then you would have two sidebar "lists" of databases with conflicting functionality.  (And in my case, this is for a bento solution where I don't want any facets.)  However this still may be a better approach, combined with just hiding the facts with CSS -- it would solve the underlying 'better results' problem and just waste some CPU cycles computing the other (unused, hidden) facets.

Reply:
> But I believe that would require actually enabling that facet (and facets in the first place) for this recommendation module to get data from it

On more digging I was happy to discover I was wrong on this.  I've re-implemented it to use the list of databases from the facets i.e. from the full results.

That has a side effect of listing way too many databases, so I will need to do another change to limit it to the first 'n' matches.

Reply:
> That has a side effect of listing way too many databases, so I will need to do another change to limit it to the first 'n' matches.

Done